server/main.go

```diff
@@ -84,37 +88,37 @@ func main() {
 			Country:     "Antarctica",
 			DisplayName: "test_user_mem",
 		})
-		service := svc.NewService(repository)
-		fmt.Println("Running app with in-memory repository")
-		Server(*port, service, repository) // Run server
-	}
 
-	proxyServerPort := os.Getenv("SERVER_PORT") // Port override
+		oauthService = &oauthSvc.Service{Repo: resourceRepository}
 
-	proxyServerPortInt, err := strconv.Atoi(proxyServerPort)
-	if err != nil {
-		log.Fatal(err)
+		fmt.Println("Running app with in-memory repository")
+	} else {
+		// If the user has set a database user or password, init the database
+		if os.Getenv("DB_USER") != "" || os.Getenv("DB_PASSWORD") != "" {
+			config.InitDB()
+		}
+
+		var err error
+		proxyPort, err = strconv.Atoi(os.Getenv("SERVER_PORT")) // Port override
```

Comment:
you can make this into one line of code

Reply:
This can't be one line, because the proxyPort variable from the outer scope would be shadowed.